### ISSUE TITLE
analyze_results.py failed with "not guess temperature from MD log ..."

### ISSUE THREAD

**Issue by Blockbastermaker** [luancarvalhomartins/PyAutoFEP#37]
(PyAutoFEP) ldd@ldd:~/Desktop $ analyze_results.py --input tutorial.tgz --units kcal --output_uncompress_directory tutorial --center_molecule=FXR_12

All available analysis will be run
[ERROR] Could not find unk_matrix.pkl file in tutorial/FXR_12-FXR_76/protein/md/rerun to read temperature from. Also, could not guess temperature from MD log in tutorial/FXR_12-FXR_76/protein/md/rerun/../lambda0/lambda.log. Set temperature argument or config option. Analysis for this system will not be run.
=================== STACK INFO ===================
  File "/home/ldd/app/PyAutoFEP/analyze_results.py", line 1508, in <module>
    verbosity=arguments.verbose)
  File "/home/ldd/app/PyAutoFEP/analyze_results.py", line 1170, in collect_from_xvg
    msg_verbosity=os_util.verbosity_level.error, current_verbosity=verbosity)
  File "/home/ldd/app/PyAutoFEP/os_util.py", line 292, in local_print
    formatted_string += '\n{:=^50}\n{}{:=^50}'.format(' STACK INFO ', ''.join(traceback.format_stack()),
=================== STACK INFO ===================
[ERROR] Could not find unk_matrix.pkl file in tutorial/FXR_12-FXR_76/water/md/rerun to read temperature from. Also, could not guess temperature from MD log in tutorial/FXR_12-FXR_76/water/md/rerun/../lambda0/lambda.log. Set temperature argument or config option. Analysis for this system will not be run.
=================== STACK INFO ===================
  File "/home/ldd/app/PyAutoFEP/analyze_results.py", line 1508, in <module>
    verbosity=arguments.verbose)
  File "/home/ldd/app/PyAutoFEP/analyze_results.py", line 1170, in collect_from_xvg
    msg_verbosity=os_util.verbosity_level.error, current_verbosity=verbosity)
  File "/home/ldd/app/PyAutoFEP/os_util.py", line 292, in local_print
    formatted_string += '\n{:=^50}\n{}{:=^50}'.format(' STACK INFO ', ''.join(traceback.format_stack()),
=================== STACK INFO ===================
[ERROR] Could not find unk_matrix.pkl file in tutorial/FXR_12-FXR_74/protein/md/rerun to read temperature from. Also, could not guess temperature from MD log in tutorial/FXR_12-FXR_74/protein/md/rerun/../lambda0/lambda.log. Set temperature argument or config option. Analysis for this system will not be run.
=================== STACK INFO ===================
  File "/home/ldd/app/PyAutoFEP/analyze_results.py", line 1508, in <module>
    verbosity=arguments.verbose)
  File "/home/ldd/app/PyAutoFEP/analyze_results.py", line 1170, in collect_from_xvg
    msg_verbosity=os_util.verbosity_level.error, current_verbosity=verbosity)
  File "/home/ldd/app/PyAutoFEP/os_util.py", line 292, in local_print
    formatted_string += '\n{:=^50}\n{}{:=^50}'.format(' STACK INFO ', ''.join(traceback.format_stack()),
=================== STACK INFO ===================
[ERROR] Could not find unk_matrix.pkl file in tutorial/FXR_12-FXR_74/water/md/rerun to read temperature from. Also, could not guess temperature from MD log in tutorial/FXR_12-FXR_74/water/md/rerun/../lambda0/lambda.log. Set temperature argument or config option. Analysis for this system will not be run.
=================== STACK INFO ===================
  File "/home/ldd/app/PyAutoFEP/analyze_results.py", line 1508, in <module>
    verbosity=arguments.verbose)
  File "/home/ldd/app/PyAutoFEP/analyze_results.py", line 1170, in collect_from_xvg
    msg_verbosity=os_util.verbosity_level.error, current_verbosity=verbosity)
  File "/home/ldd/app/PyAutoFEP/os_util.py", line 292, in local_print
    formatted_string += '\n{:=^50}\n{}{:=^50}'.format(' STACK INFO ', ''.join(traceback.format_stack()),
=================== STACK INFO ===================
[ERROR] The leg systems saved on your progress file are ['protein', 'water'] while the systems read from the input file tutorial.tgz are []. Data mismatch, cannot continue.
=================== STACK INFO ===================
  File "/home/ldd/app/PyAutoFEP/analyze_results.py", line 1585, in <module>
    msg_verbosity=os_util.verbosity_level.error)
  File "/home/ldd/app/PyAutoFEP/os_util.py", line 292, in local_print
    formatted_string += '\n{:=^50}\n{}{:=^50}'.format(' STACK INFO ', ''.join(traceback.format_stack()),
=================== STACK INFO ===================

We run analyze_results.py, got the above results. We will appreciate your kindly advise.

**Comment by luancarvalhomartins** [comment]
This error suggests that neither a unk_matrix.pkl nor a lambda.log could be read from your input. Did the run and rerun steps really run? Would you mind checking the lambda.log files?

**Comment by ZCLIU-2022** [comment]
it seems that the mpirun gmx is not going, 
gmx_mpi instead of gmx ??

**Comment by luancarvalhomartins** [comment]
Yes, the run step requires a MPI-compatible version of gmx. I'm not sure why, but I got your md.log by email and it seems fine. If your run step failed, please retry with gmx_mpi.
